src/crimson/os/seastore/scanner.cc

```diff
@@ -116,7 +116,7 @@ Scanner::scan_valid_records_ret Scanner::scan_valid_records(
     cursor.offset.offset = segment_manager.get_block_size();
   }
   auto retref = std::make_unique<size_t>(0);
-  auto budget_used = *retref;
+  auto &budget_used = *retref;
```

Comment:
I think this should fix https://tracker.ceph.com/issues/52582, am I right?

Reply:
Yep, I'll add that to the commit message.